### ISSUE TITLE
Loop guards (`when`, `case`, `|`) don't work with `for..let` loops with `yield` (or `async`) in their bodies

### ISSUE THREAD

**Issue by pepkin88** [gkz/LiveScript#1023]
Example:
```LiveScript
fn = ->*
  obj = a: 1
  for own let k, v of obj when v % 2
    yield v
```

Result:
```js
var fn;
fn = function*(){
  var obj, i$, k, v, own$ = {}.hasOwnProperty, results$ = [], fn$ = function*(k, v){
    return (yield v);
  };
  obj = {
    a: 1
  };
  for (i$ in obj) if (own$.call(obj, i$)) {
    results$.push((yield* (k = i$, v = obj[i$], v % 2 ? (fn$.call(this, k, v)) : void 8)));
  }
  return results$;
};
```

Expected result:
```js
var fn;
fn = function*(){
  var obj, i$, k, v, own$ = {}.hasOwnProperty, results$ = [], fn$ = function*(k, v){
    return (yield v);
  };
  obj = {
    a: 1
  };
  for (i$ in obj) if (own$.call(obj, i$)) {
    if (obj[i$] % 2) {
      results$.push((yield* (k = i$, v = obj[i$], v % 2 ? (fn$.call(this, k, v)) : void 8)));
    }
  }
  return results$;
};
```

I chose an example with `own`, because it also inserts its own kind of guard and it seems to be working correctly. A guard for a loop with `async` doesn't work either, but I think it's related to #1021.

Other `for..let` variants seem to be OK, those with an else block too.

**Comment by rhendric** [comment]
Also another effect of #1019.

**Comment by pepkin88** [comment]
Now I noticed, that that code doesn't work, but from different reasons. The guard `v % 2` is indeed inserted, but in the wrong spot. But, as you predicted, fix for #1019 (#1060) fixes it too.
